### PR TITLE
Feature/resign

### PR DESCRIPTION
**Pull request by yowpark**
# PR Template

## Checklist

- [ ] E2E finished.
- [ ] Local, Self Hosted and Development environment test finished.
- [ ] Local, E2E, Self Hosted, Development and Production env variable checked.
- [ ] (Backend only) DB migration file creation checked.
- [ ] **If this is a API deleted or changed**: Check side effects for API deletion or related code change.

## Related issues
- https://github.com/dogu-team/dogu/issues/496
- docs: https://www.notion.so/dogutech/9a187ae6081341778fe9ea6e9dda156b?pvs=4#e1fc0ae0d8934053a8f9c7bfe6b2a212

### CODE REVIEW

**Review by Ikalli — APPROVED** [review]
👍

File: packages/typescript-private/device-server/src/device-host/device-host.resign-app-file.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    const resignedAppPath = path.resolve(HostPaths.doguTempPath(), `resign-${uuidv4()}.ipa`);
+    const args = [shellPath, appPath, `"${identityName}"`, '-p', provisioningProfilePath, resignedAppPath].join(' ');
+
+    this.logger.info('DeviceHostResignAppFileService.resign', { args });
+    const result = await ChildProcess.exec(args, {}, this.logger);
+    this.logger.info('DeviceHostResignAppFileService.resign done', { result });
+    await fs.promises.rm(appPath, { recursive: true, force: true });
+    await fs.promises.cp(resignedAppPath, appPath, { recursive: true, force: true });
+
+    await fs.promises.rm(shellPath, { recursive: true, force: true });
```
Comment (oneofthezombies) [inline packages/typescript-private/device-server/src/device-host/device-host.resign-app-file.ts:85]: If you fail due to unexpected reasons during the re-signing process, there will be several random files.
What do you think about tying it in blocks of 
`try{ resign... } finally {if file exist then delete... }`
?

File: packages/typescript-private/device-server/src/device-host/device-host.resign-app-file.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    await fs.promises.rm(resignedAppPath, { recursive: true, force: true });
```
Comment (oneofthezombies) [inline packages/typescript-private/device-server/src/device-host/device-host.resign-app-file.ts:86]: What do you think about changing `cp` and `rm` to `mv`?

File: packages/typescript-private/device-server/src/internal/externals/cli/ideviceinstaller.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    } catch (error) {
+      this.makeAccessableSync();
+    }
+  };
+
+  private makeAccessableSync = (): void => {
+    try {
+      fs.chmodSync(HostPaths.external.libimobiledevice.ideviceinstaller(), 0o777);
+    } catch (error) {
+      const cause = error instanceof Error ? error : new Error(stringify(error));
```
Comment (oneofthezombies) [inline packages/typescript-private/device-server/src/internal/externals/cli/ideviceinstaller.ts:59]: What do you think about changing to errorify(...)?

File: packages/typescript-private/device-server/src/internal/externals/cli/ideviceinstaller.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+registerBootstrapHandler(
+  __filename,
+  async () => {
+    if (process.platform !== 'darwin') {
+      return;
+    }
+    try {
+      await fs.promises.chmod(HostPaths.external.libimobiledevice.ideviceinstaller(), 0o777);
+    } catch (error) {
+      const cause = error instanceof Error ? error : new Error(stringify(error));
```
Comment (oneofthezombies) [inline packages/typescript-private/device-server/src/internal/externals/cli/ideviceinstaller.ts:77]: What do you think about changing to errorify(...)?

**Pull request merged**